### PR TITLE
Add MerquryFK KatComp

### PR DESCRIPTION
**Pull request by mahesh-panchal**
Adds the module Merqury / KatComp which plots a k-mer comparison of two sequencing libraries.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/merquryfk/katcomp/main.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    path "versions.yml"              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def FASTK_VERSION = 'f18a4e6d2207539f7b84461daebc54530a9559b0'
+    def MERQURY_VERSION = '8f3ab706e4cf4d7b7d1dfe5739859e3ebd26c494'
```
Comment (jfy133) [inline modules/merquryfk/katcomp/main.nf:29]: Is it standard to put this here now? The example in the guidlines put it outside the process block: https://github.com/nf-core/modules/blob/master/modules/homer/annotatepeaks/main.nf

File: modules/merquryfk/katcomp/main.nf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    """
+    KatComp \\
+        $args \\
+        -T$task.cpus \\
+        ${fastk1_ktab.find{ it.toString().endsWith(".ktab") }} \\
+        ${fastk2_ktab.find{ it.toString().endsWith(".ktab") }} \\
```
Comment (jfy133) [inline modules/merquryfk/katcomp/main.nf:35]: Is it expected there would be other objects in the channel?

**Review by jfy133 — APPROVED** [review]
Couple of minor questions but LGTM

File: modules/merquryfk/katcomp/main.nf (hunk 1 above)
Comment (mahesh-panchal) [reply modules/merquryfk/katcomp/main.nf:29]: I can change them, but I prefer to have the versions within scope and as close to where they're being used. So far I've used this way with all merquryfk, genescopefk, and fastk modules. Is the homer way annotated as a standard? I feel like it's a throwback to when there were bits still outside the process like the options declaration.

File: modules/merquryfk/katcomp/main.nf (hunk 2 above)
Comment (mahesh-panchal) [reply modules/merquryfk/katcomp/main.nf:35]: Yes, there are other hidden files with the suffix `*ktab.*` which are needed for the ktab file to function.

File: modules/merquryfk/katcomp/main.nf (hunk 1 above)
Comment (mahesh-panchal) [reply modules/merquryfk/katcomp/main.nf:29]: I'll merge for now, but if this should change then I'll make a pull request to change them all

Comment (mahesh-panchal) [reply modules/merquryfk/katcomp/main.nf:29]: Sounds good. I just checked against all the modules and I'm the only one that did it differently.

**Pull request merged**